lib/rubocop/cop/layout/space_inside_parens.rb

```diff
@@ -60,6 +58,19 @@ def on_new_investigation
 
         private
 
+        def process_with_space_style(processed_source)
+          each_missing_space(processed_source.tokens) do |range|
+            add_offense(range, message: MSG_SPACE) do |corrector|
+              corrector.insert_before(range, ' ')
+            end
+          end
+          each_extraneous_space_in_empty_parens(processed_source.tokens) do |range|
```

Comment:
What's the performance impact of that second pass over the tokens? I'm wondering if we can't combine the two passes for the sake of efficiency.

Reply:
Good question. I kept them separate because the offences for empty parens is quite different than the typical cop. I'll run some benchmarks but if you're leaning towards combining them, that might be the best approach for clarity, regardless of the benchmarks.

Reply:
Okay, refactored into a single pass. Benchmark was nearly negligible but it's certainly much cleaner and more efficient now. Take a look when you get a sec.